### DISCUSSION TITLE
[Enhancement] Support earlier versions of Android & iOS

### DISCUSSION THREAD

**Discussion by eug48** [dotnet/maui#117] [General]
## Summary
Please consider at least retaining the current Xamarin.Forms support for API 19 and IOS 9.

## API Changes
I realise this is probably difficult. It would be appreciated even if older devices were supported in the core only and developers wanting to support them had to do some more work like providing down-level implementations of some aspects of their app.

## Intended Use Case
The initial plan for MAUI seems to indicate dropping support for IOS 9, meaning dropping support for quite a few devices: iPhone 4S, iPad 2, iPad (3rd generation), iPad Mini (1st generation) and iPod Touch (5th generation). (https://en.wikipedia.org/wiki/IOS_10#Supported_devices)

The initial plan to drop support for Android 4.x (KitKat and below) seems to affect ~6% of devices. (https://www.androidauthority.com/android-version-distribution-748439/)

The broader trend of dropping support for older devices particularly affects materially poorer socio-economic groups and isn't good for the environment..

Flutter by the way currently supports Android 16+ (Jelly Bean) and iOS 8+ (https://flutter.dev/docs/resources/faq#what-devices-and-os-versions-does-flutter-run-on).

Thanks for the early preview of MAUI!

**Comment by BurkusCat** [comment]
One thing to bear in mind is that MAUI comes out late 2021. Presumably, by that time, less of those devices will be in use.

**Comment by soul4soul** [comment]
+1 for not moving up minimum versions. My mobile device is still running Android 5.0. FYI so far the only app I have lost is chase mobile which dropped support over a year ago. My other apps including other finance apps continue to run on my device.

**Comment by davidortinau** [comment]
Thanks. We'd definitely like to hear from more developers, understand your customer device inventory....how many impacted in Nov 2021. 

2 new iOS versions will (likely) ship by then. 

Testing on iOS 9 devices is problematic, and as the matrix of devices grows on Android as well this has a material impact on what we can test and how long tests take to run. This is one reason that PR reviews take so long, running hundreds of tests across a spectrum of devices and versions.

**Comment by charlesroddie** [comment]
Using [statscounter](https://gs.statcounter.com/ios-version-market-share/mobile-tablet/worldwide) data for May 2020:

| iOS version     | Share |
|---------|-------|
| 13  | 78.1% |
| 12+ | 92.4% |
| 11+ | 95.0% |
| 10+ | 97.2% |

| Android version | Share |
|---------|-------|
| 10  | 35.0% |
| 9+  | 65.4% |
| 8+  | 78.0% |
| 7+  | 87.5% |
| 6+ | 91.6% |

This means that iOS 9 should be dropped with the next XF major version, not waiting for MAUI. In the timeframe of MAUI, iOS 12 and Android 7 can be the minimum.

> This is one reason that PR reviews take so long

That's way more important than supporting a tiny percentage of old devices.

> @eug48 if older devices were supported in the core only and developers wanting to support them had to do some more work

Yes interested developers should be able to fork the MAUI.IOS and MAUI.Android repos and build their own versions with support for old devices. It would be a very large effort.

**Comment by J-Swift** [comment]
In my experience, supporting Android API < 21 is a real pain that adds a nontrivial amount of duplicated code.
